src/js/utils/date-utils.js

```diff
@@ -0,0 +1,292 @@
+};
+
+
+(function(define) {
+    'use strict';
+    define([
+        'jquery',
+        'edx-ui-toolkit/js/utils/string-utils',
+        'moment',
+        'momenttimezone'
+    ], function(
+        $,
+        StringUtils,
+        MomentLib,
+        MomentTZ
+        ) {
+        /**
+         * Create a formatted, locale-aware string representation of a datetime.
+         *
+         * These functions are meant to act as a general localization function
+         * for UTC datetime objects, creating platform-friendly strings in the user's
+         * local timezone formatted to the user's preferred language
+         *
+         * Most of the heavy lifting is done by the 'moments' js library,
+         * with the moment-with-locales and moment-timezones add-ons
+         *
+         * by sending preferences data, even in an incomplete state, we can generate the appropriate
+         * datetime for surfacing on the frontend
+         *
+         * for example, if you're a French speaker currently accessing the site in Rangoon,
+         * who you have not set your timezone preferences:
+         *
+         * var context = {
+         *    datetime: '2016-10-14 08:00:00+00:00',
+         *    language: "fr",
+         *    }
+         * DateUtils.localize(context)
+         *
+         * which will render: '14 oct. 2016 14:30 MMT'
+         *
+         */
+        var DateUtils, DateObject, UserTimeZone, LocalizedTime, TimeDisplay, localize;
+
+        /**
+         * convert sent string to mutable 'moment' datetime object
+         *
+         * @param {string} datetime, can be just a date,
+         * or a complex date/time string with TZ offset. This is fairly open, and
+         * many strings will satisfy this (01-01-16 and Jan 1, 2016 will both work)
+         *
+         * @returns {object} a moment.js datetime object in UTC
+         *
+        */
+        DateObject = function(datetime) {
+            if (datetime === undefined || datetime === '') {
+                return '';
+            }
+            return MomentLib(datetime).utc();
+        };
+
+        /**
+         * attempt to determine user's timezone
+         * from preferences, then browser settings
+         *
+         * @param {string} timezone representation, 'Region/Population_Center' e.g. 'America/New_York'.
+         * If this is None or undefined, we attempt to determine the TZ from system/browser settings
+         *
+         * @return {string} either the entered or determined
+         * timezone, or (in the case of default) UTC
+         *
+         */
+        UserTimeZone = function(timezone) {
+            // linter wants a different variable
+            var returnzone;
+            if (timezone === undefined || timezone === 'None') {
+                returnzone = MomentTZ.tz.guess();
+            } else {
+                returnzone = timezone;
+            }
+            if (returnzone === undefined) {
+                return 'UTC';
```

Comment:
You can define `returnzone` as `'UTC'` initially, then just let it be re-assigned (or not) as needed.